ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/PaywallDataExtensions.kt

```diff
@@ -0,0 +1,99 @@
+    val locale = LocaleListCompat.getDefault()[0].toString()
+
+    return PaywallData(
+        templateName = PaywallData.defaultTemplate,
+        config = PaywallData.Configuration(
+            packages = packageIdentifiers,
+            images = PaywallData.Configuration.Images(
+                background = PaywallData.defaultBackgroundImage,
+                icon = PaywallData.defaultAppIconPlaceholder,
+            ),
+            colors = PaywallData.defaultColors,
+            blurredBackgroundImage = true,
+            displayRestorePurchases = true,
+        ),
+        localization = mapOf(locale to PaywallData.defaultLocalization), // TODO-PAYWALLS: test this
+        assetBaseURL = PaywallData.defaultTemplateBaseURL, // TODO-PAYWALLS: we need to load a resource not a URL
+        revision = PaywallData.revisionID,
+    )
+}
+
+// region Internal defaults
+
+internal val PaywallData.Companion.defaultTemplate: String // TODO-PAYWALLS: use enum
+    get() = "2"
+
+internal val PaywallData.Companion.defaultAppIconPlaceholder: String // TODO-PAYWALLS: use real icon
+    get() = "revenuecatui_default_paywall_app_icon"
+
+internal val PaywallData.Companion.revisionID: Int
+    get() = -1
+
+// endregion
+
+// region Private defaults
+
+private val PaywallData.Companion.defaultBackgroundImage: String
+    get() = "background.jpg"
+
+private val PaywallData.Companion.defaultTemplateBaseURL: URL // TODO-PAYWALLS: use real URL
+    get() = URL("https://")
+
+private val PaywallData.Companion.defaultColors: PaywallData.Configuration.ColorInformation
+    @Composable
+    @ReadOnlyComposable
+    get() {
+        return PaywallData.Configuration.ColorInformation(
+            light = getCurrentColors(background = PaywallColor(colorInt = Color.White.toArgb())),
+            dark = getCurrentColors(background = PaywallColor(colorInt = Color.Black.toArgb())),
+        )
+    }
+
+private val PaywallData.Companion.defaultLocalization: PaywallData.LocalizedConfiguration
+    get() = PaywallData.LocalizedConfiguration(
+        title = "{{ app_name }}",
+        callToAction = "Continue",
+        offerDetails = "{{ total_price_and_per_month }}",
+        offerDetailsWithIntroOffer = "Start your {{ sub_offer_duration }} trial, then {{ total_price_and_per_month }}.",
+    )
+
+// endregion
+
+@Composable
+@ReadOnlyComposable
+private fun getCurrentColors(
+    background: PaywallColor,
+): PaywallData.Configuration.Colors {
+    return PaywallData.Configuration.Colors(
+        background = background,
+        text1 = MaterialTheme.colors.primary.asPaywallColor(),
+        callToActionBackground = MaterialTheme.colors.secondary.asPaywallColor(),
+        callToActionForeground = background,
+        accent1 = MaterialTheme.colors.secondary.asPaywallColor(),
+        accent2 = MaterialTheme.colors.primary.asPaywallColor(),
+    )
+}
+
+private fun Color.asPaywallColor(): PaywallColor = PaywallColor(colorInt = this.toArgb())
```

Comment:
Does this support dark mode too?

Reply:
Unlike iOS, in Android, Color doesn't have "context" of the light/dark mode

In this case, I am getting current colors (see `getCurrentColors`) which accesses `MaterialTheme.colors`, which changes when the user changes between light and dark mode (I need to test this but that's what I understand)

Reply:
Makes sense 👍🏻